renovate.json

```diff
@@ -31,8 +27,18 @@
       "groupName": "golang.org/x"
     },
     {
+        "_description": "Ignore pin and digest upgrades. We only want published releases of the SDK to trigger upgrades",   
       "matchPackageNames": ["go.opentelemetry.io/otel/**"],
-      "allowedVersions": "/^v[0-9]+\\.[0-9]+\\.[0-9]+/"
+      "enabled": true,
+      "ignore": {
+        "matchUpdateTypes": ["pin", "digest"]
```

Comment:
What is a pin?

Reply:
It looks like it's a dependency set at an explicit version.
https://docs.renovatebot.com/dependency-pinning/

Pinning isn't something we can really do with Go. But just to be same, maybe we should actually remove that match.

Reply:
@pellared I would be happy to! sorry for messing it up , i'm  trying but my PR keeps getting tested here instead can you please give me some pointers on how to test directly on my fork repo . thanks

Reply:
@dmathieu Did we succeeded in fixing this issue i would love a feedback if there is anything else i need to do i will be happy to! Thanks

Reply:
There are still open questions, in this very thread that need resolving.
Especially, the tricky thing here is that we can't really confirm we "succeeded", since there's no way to validate the behavior of the config until it's merged and runs. Hence @pellared's question to try this in a fork, or another repository.

Reply:
@Victorsesan, you need to configure Renovate on your fork to check if it will work as expected.

Reply:
@pellared @dmathieu Are these the expected test we need to see? Thanks for the time
![Screenshot 2024-10-24 004430](https://github.com/user-attachments/assets/a0b253ce-7145-4b8a-b605-051a2efaf4ab)
![Screenshot 2024-10-24 004348](https://github.com/user-attachments/assets/1e4c9414-077e-4930-90f3-39e0c4c8f26e)
![Screenshot 2024-10-24 004059](https://github.com/user-attachments/assets/436d2c93-58df-47b9-95d8-429b841e1a79)

Reply:
They would be, yes. But in your screenshot, they appear to say the configuration file is invalid.

Reply:
@dmathieu Finally succeeded in making renovate work as expected after a long try and errors, i had to make some minor changes for it to pass the test which I'm not sure if its good or bad but i mentioned all the errors i faced with the solution and modifications i did in my recent commit which i will be happy if you can check it out and let me know if we are in the right track. I would love to succeed in working on this . Thank you all for your time.
![Screenshot 2024-10-25 010858](https://github.com/user-attachments/assets/890c01f0-09a7-40c9-952d-0a64ea4f9fa6)
![Screenshot 2024-10-25 010932](https://github.com/user-attachments/assets/ef3c62b0-d96c-4da9-a585-d09665c54bd4)

Reply:
This shows the config is valid. But it doesn't seem to validate that commit upgrades would be skipped, while releases would be bumped?

Reply:
> hows the config is valid. But it doesn't seem to validate that commit upgrades would be skipped, while releases would be bumped

@dmathieu Sorry i missed that, is this last commit good i have added something to fix it. I was just worried with mend not accepting the package rules even when they are good sometimes which made me lost track validating the skip rule and rather focus on having renovate pass the test, but let me know if the few changes i made does it and it did also pass the test. thanks